### ISSUE TITLE
Error: Failed to load native binding

### ISSUE THREAD

**Issue by dghez** [nuxt/eslint#573]
I'm installing it on a fresh new project and i get

```js
[Error - 12:07:22] An unexpected error occurred:
[Error - 12:07:22] Error: Failed to load native binding
    at Object.<anonymous> (/Users/-----------/node_modules/unrs-resolver/index.js:372:11)
    at Module._compile (node:internal/modules/cjs/loader:1484:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1564:10)
    at Module.load (node:internal/modules/cjs/loader:1295:32)
    at Module._load (node:internal/modules/cjs/loader:1111:12)
    at c._load (node:electron/js2c/node_init:2:16955)
    at cjsLoader (node:internal/modules/esm/translators:350:17)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:286:7)
    at ModuleJob.run (node:internal/modules/esm/module_job:234:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:473:24)
```

I tried with different node versions (`20.10.0`, `20.17.0`, `20.18.0`) always getting the same result.
Apparently the issue is coming from `unrs-resolver` that is a dependency of 

```js
➜ .... git:(develop) ✗ npm ls unrs-resolver

nuxt-app@ /Users/...............
└─┬ @nuxt/eslint@1.3.0
  └─┬ @nuxt/eslint-config@1.3.0
    └─┬ eslint-plugin-import-x@4.10.2
      └── unrs-resolver@1.4.1
```

I tried even different versions of `@nuxt/eslint`, from 0.7.3 to this one, same issue.
Any clue?

Thanks :)

**Comment by dghez** [comment]
Apparently tho is a fact that vscode/cursor node version doesn't support the `unrs-resolver`
Setting in the settings.json the node instance to use (same as editor but little different) fixed the issue, see below

![Image](https://github.com/user-attachments/assets/438a2d86-45c2-47e4-932e-ef25b9ac67a6)

**Comment by baijii** [comment]
I had the same issue with Node 22 and Nuxt 3.16, but only in the CI. Downgrading from `@nuxt/eslint: "^1.3.0"` to `@nuxt/eslint: "~1.2.0"` did the trick for me (for now)

**Comment by sneakylenny** [comment]
I get the same but adding a runtime wouldn't fix it for me. But in my case I required a different fix:

### Rundown

Here's a log for more context:

<details>
  <summary>Click to expand</summary>

  ```
[Info  - 8:40:01 AM] ESLint server is starting.
[Info  - 8:40:02 AM] ESLint server running in node v20.18.3
[Info  - 8:40:02 AM] ESLint server is running.
[Info  - 8:41:32 AM] ESLint library loaded from: /Users/tim/Projects/myproject/node_modules/eslint/lib/api.js
[Error - 8:41:33 AM] Calculating config file for file:///Users/tim/Projects/myproject/example.js) failed.
Error: Failed to load native binding
    at Object.<anonymous> (/Users/tim/Projects/myproject/node_modules/.pnpm/unrs-resolver@1.7.2/node_modules/unrs-resolver/index.js:372:11)
    at Module._compile (node:internal/modules/cjs/loader:1484:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1564:10)
    at Module.load (node:internal/modules/cjs/loader:1295:32)
    at Module._load (node:internal/modules/cjs/loader:1111:12)
    at c._load (node:electron/js2c/node_init:2:16955)
    at cjsLoader (node:internal/modules/esm/translators:350:17)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/translators:286:7)
    at ModuleJob.run (node:internal/modules/esm/module_job:234:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:473:24)
  ```
</details>

It's seems like it's caused by `unrs-resolver` as it points to "unrs-resolver@1.7.2/node_modules/unrs-resolver/index.js:372:11". In this file there's a couple of checks for architecture.

In my case the dependencies were installed from a Docker container. My machine is a Mac M3 (ARMv8-A) and the node container is a Linux aarch64 machine. So I guess there's what's causing the mismatch.

### TL;DR

After installing the dependencies locally on my machine the error went away.

#### Extra:

Besides this I had some issues using `@nuxt/eslint` as somehow the ESLint command was not available, so I added `eslint` as a dependency and it seemed to fix that.

**Comment by The-Best-Joke** [comment]
I'm working on an Apple Silicon Mac, I just recently had this problem pop up within my VSCode editor setup, and none of the detailed solutions above worked.

Neither setting a previous node version and establishing its use with a `.nvmrc`, nor reverting to a previous `@nuxt/eslint` version, nor pruning the entire pnpm cache and rebuilding the `node_modules` and `pnpm-lock.yaml` file. Nor installing all the dependencies locally. Nothin'.

HOWEVER, what did work, surprisingly and annoyingly, was to simply remove the wrapper `withNuxt` function from the `eslint.config.js` file, running the Nuxt dev server, restarting the ESLint server, adding BACK the `withNuxt` function to the config file, and restarting the ESLint server once more.

It seems that whatever the issue with the `unrs-resolver` is, it only pops up at first load on Nuxt projects that haven't already built their caching. Once that's setup the issue seems to disappear.

Not an ideal solution but at least it's a workaround for now.
